File: 09/main.py
def check(data, x, y):
    if x > 0 and data[x-1][y] <= data[x][y]:
        return None
    if x < len(data)-1 and data[x+1][y] <= data[x][y]:
        return None
    if y > 0 and data[x][y-1] <= data[x][y]:
        return None
    if y < len(data[0])-1 and data[x][y+1] <= data[x][y]:
        return None
    return data[x][y] + 1


with open("input.txt") as f:
    data = [[int(c) for c in line]
            for line in f.read().split("\n")]


out = 0
for x in range(len(data)):
    for y in range(len(data[0])):
        c = check(data, x, y)
        if c is not None:
            out += c

print(out)

File: 09/main.cpp
#include "../common.h"
#include <fstream>
#include <iostream>

using namespace std;

int check(vector<string> data, int x, int y) {
    if((x > 0) && (data[x-1][y] <= data[x][y]))
        return -1;
    if((x < data.size()-1) && (data[x+1][y] <= data[x][y]))
        return -1;
    if((y > 0) && (data[x][y-1] <= data[x][y]))
        return -1;
    if((y < data[0].length()-1) && (data[x][y+1] <= data[x][y]))
        return -1;
    return data[x][y] - '0' + 1;
}

int main()
{
    ifstream fp("input.txt");
    vector<string> lines = read_lines(fp);
    int out = 0;

    for (int x = 0; x < lines.size(); x++) {
        for (int y = 0; y < lines[0].length(); y++) {
            int c = check(lines, x, y);
            if (c != -1) {
                out += c;
            }       
        }
    }
    cout << out << "\n";
}